### ISSUE TITLE
Expected remote cache functionality

### ISSUE THREAD

**Issue by arvidboivie** [nrwl/nx#28936]
### Documentation issue

- [ ] Reporting a typo
- [ ] Reporting a documentation bug
- [x] Documentation improvement
- [x] Documentation feedback

### Is there a specific documentation page you are reporting?

https://nx.dev/nx-enterprise/powerpack/custom-caching (Or somewhere adjacent)

### Additional context or description

When implementing the Powerpack S3 Remote cache, I came across [this post](https://github.com/nrwl/nx/issues/28776#issuecomment-2455519637) which makes me wonder about the intended functionality of the remote cache. Is it supposed to work only in CI environments?

From early testing it seems that it neither pushes to or pulls from the remote cache when running tasks locally, which is not what I would have expected. There are no info or error messages displayed regarding caching (local cache still works as expected). There's also no information in the docs that I've been able to find regarding this.

**Note:** Caching works in our CI environment, so this doesn't seem to be a configuration issue?

**Expected behaviour:**

Remote cache will read/write regardless if task is run on developer machine or in CI environment.
If this isn't the intended behaviour, I would argue that the documentation needs to make it clear that that is the case.

**Comment by arvidboivie** [comment]
Thanks for the info!

I agree that having this as an option (or the default functionality) would be great. I guess you could set up a `.env` file with `CI=true` that Nx would read ([like this](https://nx.dev/recipes/tips-n-tricks/define-environment-variables#set-environment-variables)).

But there's probably more tools looking for the CI variable, so it might have other unintended effects as well when turning it on locally.

**Comment by giltayar** [comment]
Exactly. Would NX be recipient to a PR that adds the ability to configure it via `nx.json` or a CLI argument?

**Comment by BenjaminBalleux** [comment]
Setting CI=1 does cause some issues for me. It makes nx reset and some targets unable to run. I'm not really surprised about this, my environment is not really close of a CI environment.

I can not really understand this change, having remote caching on local for developers is a core feature of it..
This was totally functional with legacy custom caching implentation.

**Comment by Maus3rVonDutch** [comment]
To chime in, in the deprecated custom implementation of `nx-remotecache-azure`, we get below log messages per action(test/lint etc), which I found very helpful, please consider it as an (configurable) option, or at least some other logging so you know the caching actually works.

```
------------------------------------------------------------------------------
Remote cache hit: Azure Blob Storage
File: 3187715821839764177.tar.gz
------------------------------------------------------------------------------

------------------------------------------------------------------------------
Stored output to remote cache: Azure Blob Storage
File: 7797811707198499742.tar.gz
------------------------------------------------------------------------------
```

**Comment by Cammisuli** [comment]
Hey @arvidboivie we recently just released an update to powerpack that allows users to set the type of operations or modes that the cache operate in with local or in CI contexts. By default, we now enable local runs to read and write to the remote cache. 

You can read more about it here:
https://nx.dev/nx-api/powerpack-s3-cache#cache-modes

@Maus3rVonDutch thanks for your input, I'll create a task internally to create additional logging for the cache hits.
